### PR TITLE
SHARD-916 Workflow addition to build and push docker image

### PR DESCRIPTION
**Pull request by arhamj**
https://linear.app/shm/issue/SHARD-916

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# syntax=docker/dockerfile:1
+
+## global args
+ARG NODE_VERSION=18.16.1
+
+FROM node:${NODE_VERSION}
```
Comment (aniketdivekar) [inline Dockerfile]: have we tried using alpine image? that should reduce the image size by a considerable amount. not a blocker for this PR but we should check this.

Comment (arhamj) [reply Dockerfile]: Thanks for the suggestion this made the image small by 1 GB. Tested and updated

**Review by aniketdivekar — DISMISSED** [review]

**Review by aniketdivekar — DISMISSED** [review]

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+on:
+  push:
+    branches: ['dev']
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag for the Docker image'
+        required: false
+        default: 'latest' # Default tag if none is provided
```
Comment (chrypnotoad) [inline .github/workflows/docker.yml]: this input isn't being used when the tag is set

Comment (arhamj) [reply .github/workflows/docker.yml]: Updated

**Review by chrypnotoad — APPROVED** [review]

**Pull request merged**